src/pykeen/sampling/basic_negative_sampler.py

```diff
@@ -31,31 +34,83 @@ class BasicNegativeSampler(NegativeSampler):
         num_negs_per_pos=dict(type=int, low=1, high=100, q=10),
     )
 
-    def sample(self, positive_batch: torch.LongTensor) -> torch.LongTensor:
+    def __init__(
+        self,
+        triples_factory: TriplesFactory,
+        num_negs_per_pos: Optional[int] = None,
+        filtered: bool = False,
+        corruption_scheme: Set[str] = None,
+    ) -> None:
+        """Initialize the negative sampler with the given entities.
+
+        :param triples_factory: The factory holding the triples to sample from
+        :param num_negs_per_pos: Number of negative samples to make per positive triple. Defaults to 1.
+        :param filtered: Whether proposed corrupted triples that are in the training data should be filtered.
+            Defaults to False.
+        :param corruption_scheme: What sides ('h', 'r', 't') should be corrupted. Defaults to head and tail ('h', 't').
+        """
+        super().__init__(
+            triples_factory=triples_factory,
+            num_negs_per_pos=num_negs_per_pos,
+            filtered=filtered,
+        )
+        self.corruption_scheme = corruption_scheme or ('h', 't')
```

Comment:
I think `Collection` is probably the best type for the input type. You could also change it to
```suggestion
        self.corruption_scheme = set(corruption_scheme or ('h', 't'))
```

If you want the `self.corruption_scheme` to be a set object always, regardless of the input type

Reply:
Doesn't matter as long as it's iterable. So I'll `Collection` as @mberr suggested.